pkg/jobs/hive/hive.go

```diff
@@ -329,8 +334,268 @@ func UpgradeCluster(client clientv1.Client, clusterName string, curator *cluster
+	managedclusterview := &managedclusterviewv1beta1.ManagedClusterView{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      clusterName,
+			Namespace: clusterName,
+			Labels: map[string]string{
+				MCVUpgradeLabel: clusterName,
+			},
+		},
+		Spec: managedclusterviewv1beta1.ViewSpec{
+			Scope: managedclusterviewv1beta1.ViewScope{
+				Group:     "config.openshift.io",
+				Kind:      "ClusterVersion",
+				Name:      "version",
+				Namespace: "",
+				Version:   "v1",
+			},
+		},
+	}
+
+	mcview := managedclusterviewv1beta1.ManagedClusterView{}
+	if err := client.Get(context.TODO(), types.NamespacedName{
+		Namespace: clusterName,
+		Name:      clusterName,
+	}, &mcview); err != nil {
+
+		klog.V(2).Info("Create managedclusterview " + clusterName)
+		if err := client.Create(context.TODO(), managedclusterview); err != nil {
+			return err
+		}
+	}
+
+	resultmcview := managedclusterviewv1beta1.ManagedClusterView{}
+	if err := waitForMCV(client, clusterName, clusterName, &resultmcview, getErr); err != nil {
+		return err
+	}
+
+	resultClusterVersion := resultmcview.Status.Result
+	clusterVersion := map[string]interface{}{}
+
+	if resultClusterVersion.Raw != nil {
+		err := json.Unmarshal(resultClusterVersion.Raw, &clusterVersion)
+		utils.CheckError(err)
+	} else {
+		return getErr
+	}
+
+	currentOCPVer := clusterVersion["status"].(map[string]interface{})["desired"].(map[string]interface{})["version"].(string)
+	desiredSemVer, err := semver.Make(currentOCPVer)
+	if err != nil {
+		return err
+	}
+
+	ackString := "ack-" + strconv.Itoa(int(desiredSemVer.Major)) + "." + strconv.Itoa(int(desiredSemVer.Minor)) +
+		"-kube-1." + strconv.Itoa(int(desiredSemVer.Minor+14)) + "-api-removals-in-" +
+		strconv.Itoa(int(desiredSemVer.Major)) + "." + strconv.Itoa(int(desiredSemVer.Minor+1))
+
+	cmData := configMap["data"]
+	if cmData != nil {
+		// "kube-1.{{ '%02d' | format((ocp_minor_version.split('.')[1] | int) + 14"
+		cmData.(map[string]interface{})[ackString] = "true"
+	} else {
+		configMap["data"] = map[string]interface{}{
+			ackString: "true",
+		}
+	}
+
+	// update managed cluster with acknowledge config to allow upgrade
+	var updateConfigMap runtime.RawExtension
+	if configMap != nil {
+		b, err := json.Marshal(configMap)
+		utils.CheckError(err)
+		updateConfigMap.Raw = b
+	}
+	klog.V(2).Info("Create managedclusteraction to update configmap " + clusterName + "admack")
+	ocpConfigMCA := &managedclusteractionv1beta1.ManagedClusterAction{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      clusterName + "admack",
+			Namespace: clusterName,
+		},
+		Spec: managedclusteractionv1beta1.ActionSpec{
+			ActionType: managedclusteractionv1beta1.UpdateActionType,
+			KubeWork: &managedclusteractionv1beta1.KubeWorkSpec{
+				Resource:       "configmaps",
+				Name:           "admin-acks",
+				Namespace:      "openshift-config",
+				ObjectTemplate: updateConfigMap,
+			},
+		},
+	}
+	if err := client.Create(context.TODO(), ocpConfigMCA); err != nil {
```

Comment:
I am expecting this is a bug since it's Create only and no Get to make sure it exists like the other similar flows above?